src/App.js

```diff
@@ -1,23 +1,37 @@
-import { useState } from "react"
+import { useState, useEffect } from "react"
 import TodoList from "./TodoList"
 import AddTodoForm from "./AddTodoForm"
 
+function useSemiPersistentState() {
+  const [todoList, setTodoList] = useState(
+    JSON.parse(localStorage.getItem("savedTodoList"))
+      ? JSON.parse(localStorage.getItem("savedTodoList"))
+      : []
```

Comment:
good catch, on setting the initial state! 🏅 

You can also set the initial state like this:
`useState(JSON.parse(localStorage.getItem("savedTodoList")) || [])`

^ This is saying try to run the `JSON.parse(localStorage.getItem("savedTodoList"))`, if no value returned or `undefined`. run `[]`.

Reply:
Thank you!